ckanext/datastore/interfaces.py

```diff
@@ -232,3 +235,82 @@ def update_datastore_info_field(
         in the data dictionary page.
         """
         return field
+
+
+class IDatastoreDump(interfaces.Interface):
+    """
+    Allows pluggable dump formats
+    """
+    def __init__(self, output: Optional[Union[StringIO, BytesIO]]=None, columns: Optional[list[str]]=None):
+        self.output = output
+        if columns:
+            self.id_col = columns[0] == '_id'
+            if self.id_col:
+                columns = columns[1:]
+        self.columns = columns
+
+
+    def __call__(self, output: Optional[Union[StringIO, BytesIO]]=None, columns: Optional[list[str]]=None):
+        self.__init__(output, columns)
+        return self
```

Comment:
If we call an instance the `__init__` is called again with new parameters? That's an unusual pattern. There's almost no cost to creating a new instance, and if there was a normal method called `reset` or `set_parameters` would be better.